### PR TITLE
Allow JCL to be submitted from stdin

### PR DESCRIPTION
**Pull request by crshnburn**
While working on a project it became clear that being able to supply the JCL to submit on stdin would be useful as it would allow the Zowe-cli program to run as part of a unix pipeline.

The change adds a new `stdin` option which can be used in place of the existing data set or local file options.

Example of usage:
`mustache config.json template.jcl | zowe jobs sub stdin --vasc`

Signed-off-by: Andrew Smithson <smithson@uk.ibm.com>

### CODE REVIEW

**Comment by zFernand0** [comment]
Hey @crshnburn, 
This looks great! Thanks for adding this functionality.
Would you mind updating that one snapshot with the new `stdin` option?

File: packages/zosjobs/src/cli/submit/stdin/stdin.definition.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    type: "command",
+    summary: "Submit a job read from standard in",
+    description: "Submit a job (JCL) passed to the command via the stdin stream. " +
+        "The command presents errors verbatim from the z/OSMF Jobs REST endpoints. " +
+        "For more information about z/OSMF Jobs API errors, see the z/OSMF Jobs API REST documentation.",
+    handler: __dirname + "/../Submit.shared.handler",
+    options: ([
+        {
+            name: "view-all-spool-content", aliases: ["vasc"],
+            description: "View all spool content for specified job ID",
```
Comment (tucker01) [inline packages/zosjobs/src/cli/submit/stdin/stdin.definition.ts]: This description doesn't look quite right here?

**Comment by tucker01** [comment]
Also, would it be possible to add a CLI system test for this?

File: packages/zosjobs/src/cli/submit/stdin/stdin.definition.ts (hunk 1 above)
Comment (crshnburn) [reply packages/zosjobs/src/cli/submit/stdin/stdin.definition.ts]: I've updated the description, originally it had been copied from the local-file definition but this suffers from the same issues, so have updated them to match the dataset command.

**Comment by crshnburn** [comment]
> Also, would it be possible to add a CLI system test for this?

Yep, I'll add a system test, plus unit test for the new definition.

**Comment by tucker01** [comment]
Thanks for adding the tests.

**Comment by tucker01** [comment]
Actually, unless I missed it, looks like a help test was added, but no system/CLI test script to submit a job from stdin. Mind adding one more? You can model it after the tests found here: https://github.com/zowe/zowe-cli/blob/master/packages/zosjobs/__tests__/__system__/cli/submit/local-file/cli.zos-jobs.submit.local-file.system.test.ts

**Comment by crshnburn** [comment]
I've added some tests which mirror the local file tests but pipe the JCL into the Zowe CLI and use the stdin option.

**Comment by tucker01** [comment]
Awesome. Thanks for contributing! Once, the merge with base branch build competes. I'll approve and merge.

**Pull request merged**